src/common/datasource/src/file_format/csv.rs

```diff
@@ -297,4 +307,226 @@ mod tests {
+                1024,
+                1,
+                &format,
+            )
+            .await
+            .unwrap();
+
+            assert_eq!(rows, 9);
+
+            // Verify compressed file was created and has content
+            assert!(compressed_file_path.exists());
+            let file_size = std::fs::metadata(&compressed_file_path).unwrap().len();
+            assert!(file_size > 0);
+
+            // Verify the file is actually compressed
+            let file_content = std::fs::read(&compressed_file_path).unwrap();
+            // Compressed files should not start with CSV header
+            // They should have compression magic bytes
+            match compression_type {
+                CompressionType::Gzip => {
+                    // Gzip magic bytes: 0x1f 0x8b
+                    assert_eq!(file_content[0], 0x1f, "Gzip file should start with 0x1f");
+                    assert_eq!(
+                        file_content[1], 0x8b,
+                        "Gzip file should have 0x8b as second byte"
+                    );
+                }
+                CompressionType::Bzip2 => {
+                    // Bzip2 magic bytes: 'BZ'
+                    assert_eq!(file_content[0], b'B', "Bzip2 file should start with 'B'");
+                    assert_eq!(
+                        file_content[1], b'Z',
+                        "Bzip2 file should have 'Z' as second byte"
+                    );
+                }
+                CompressionType::Xz => {
+                    // XZ magic bytes: 0xFD '7zXZ'
+                    assert_eq!(file_content[0], 0xFD, "XZ file should start with 0xFD");
+                }
+                CompressionType::Zstd => {
+                    // Zstd magic bytes: 0x28 0xB5 0x2F 0xFD
+                    assert_eq!(file_content[0], 0x28, "Zstd file should start with 0x28");
+                    assert_eq!(
+                        file_content[1], 0xB5,
+                        "Zstd file should have 0xB5 as second byte"
+                    );
+                }
+                _ => {}
+            }
+
+            // Verify the compressed file can be decompressed and content matches original data
+            let store = test_store("/");
+            let schema = Arc::new(
+                CsvFormat {
+                    compression_type,
+                    ..Default::default()
+                }
+                .infer_schema(&store, compressed_file_path_str)
+                .await
+                .unwrap(),
+            );
+            let csv_source = CsvSource::new(true, b',', b'"')
+                .with_schema(schema.clone())
+                .with_batch_size(8192);
+
+            let mut stream = file_to_stream(
+                &store,
+                compressed_file_path_str,
+                schema.clone(),
+                csv_source.clone(),
+                None,
+                compression_type,
+            )
+            .await
+            .unwrap();
```

Comment:
We can simplify it to:
```suggestion
            let mut stream = file_to_stream(
                &store,
                compressed_file_path_str,
                schema.clone(),
                csv_source.clone(),
                None,
                compression_type,
            )
            .await
            .unwrap();
            let batches = stream.try_collect::<Vec<_>>().await.unwrap();
            let pretty_print = pretty_format_batches(&batches).unwrap().to_string();
            let expected = r#"+----+---------+-------+
| id | name    | value |
+----+---------+-------+
| 1  | Alice   | 10.5  |
| 2  | Bob     | 20.3  |
| 3  | Charlie | 30.7  |
| 4  | David   | 40.1  |
| 5  | Eva     | 50.2  |
| 6  | Frank   | 60.3  |
| 7  | Grace   | 70.4  |
| 8  | Henry   | 80.5  |
| 9  | Ivy     | 90.6  |
+----+---------+-------+"#;
            assert_eq!(expected, pretty_print);
```